src/utils.zig

```diff
@@ -2221,11 +2221,23 @@ fn CastFunctionSelfToAnyopaqueType(Fn: type) type {
 	typeInfo.@"fn".params = params[0..];
 	return @Type(typeInfo);
 }
+fn CastFunctionSelfToOptionalAnyopaqueType(Fn: type) type {
```

Comment:
What would an optional self parameter mean, an how would it ever be useful?
Don't add functions that we don't need.